libsolidity/ast/ASTUtils.h

```diff
@@ -33,4 +35,7 @@ VariableDeclaration const* rootConstVariableDeclaration(VariableDeclaration cons
 /// Returns true if the constant variable declaration is recursive.
 bool isConstantVariableRecursive(VariableDeclaration const& _varDecl);
 
+/// Locates closest AST node match within a SourceUnit given its absolute file offset or nullptr if none matched.
+ASTNode const* locateASTNode(int _pos, SourceUnit const& _sourceUnit);
```

Comment:
The names here could be a bit more self-explanatory:
```suggestion
ASTNode const* closestASTNode(int _positionInFile, SourceUnit const& _sourceUnit);
```

Reply:
Actually, the docstring is wrong, it does not locate the closest node, just any node covering that location, so maybe this would be better:

```suggestion
ASTNode const* anyASTNodeAt(int _positionInFile, SourceUnit const& _sourceUnit);
```

Reply:
It returns the innermost ast node that covers the location.

Reply:
If it does then it's really not clear why. I don't see anything that would choose specifically an innermost match:

```c++
	bool visitNode(ASTNode const& _node) override
	{
		if (!_node.location().containsOffset(m_offsetInFile))
			return false;
		m_innermostMatch = &_node;
		return true;
	}
```

Is this because of the visiting order? If so, this at least needs a comment.

Reply:
inner-most as in recursively descending the AST until `containsOffset` fails. If it starts to fail you know you have reached the inner-most AST node (in `m_innermostMatch` that still contains the input file offset. What text would you like to see as comment here?

Reply:
Looks like I missed the `!` and misinterpreted the code. I thought that you're actually taking the first one you find. Does not help that between `return false` and `return true` it's not that obvious which one makes the visitor stop (maybe we should have constants for them, e.g. `AST_CONTINUE` and `AST_STOP`?).

How about this:
``` c++
// In the AST parent location always covers the whole child location. Parent is visited first so to
// get the inntermost node we simply take the last one that still contains the offset.
if (!_node.location().containsOffset(m_offsetInFile))
	return false;
...
```

Reply:
> Does not help that between return false and return true it's not that obvious which one makes the visitor stop

That's not a problem of this PR but rather how we stop the visitors in general. All visitor implementations are doing the same return-logic. And we (a few at least) agreed on `bool` not being the best choice as return type for that exact reason. I'm all in for changing that (in another PR) for our Visitor API in general.

Reply:
Yeah, I did not mean that we should change that now, just saying that this is another factor that makes it easier to misinterpret the code.